### ISSUE TITLE
Message ordering seems broken starting with 5.0.1

### ISSUE THREAD

**Issue by ls-urs-keller** [GoogleCloudPlatform/spring-cloud-gcp#2649]
**Describe the bug**
When switching from 5.0.0 to 5.0.1 there seems to be a problem with message ordering.
The problem is present in 5.0.1, 5.0.2, 5.0.3 and 5.0.4.

We are using Spring 3.2.3, Spring cloud 2023.0.0, java 21

**Sample**
I have created a demonstrator project here: https://github.com/ls-urs-keller/pubsub-message-ordering-problem
It has multiple maven profiles that allow you to run:
- the broken version (5.0.1)
- the working version (5.0.0)
- a version with a workaround (5.0.1 + google-cloud-pubsub-bom pinned to version 1.125.13 which was used in 5.0.0)

Please refer to the [readme.md ](https://github.com/ls-urs-keller/pubsub-message-ordering-problem?tab=readme-ov-file#demonstrator-message-ordering-problem) of the project

**Comment by burkedavison** [comment]
This issue looks very similar to a recently resolved pubsub client issue: https://github.com/googleapis/java-pubsub/issues/1889

@ls-urs-keller : When I tried replicating with your project using the production service (not the emulator), I got the following error due to the `@Autowired PubSubTemplate`:
```
java.lang.IllegalStateException: Cannot publish a message with an ordering key when message ordering is not enabled in the Publisher client. Please create a Publisher client with setEnableMessageOrdering(true) in the builder.
```
When I manually created the PubSubTemplate with a PublisherFactory that set `factory.setEnableMessageOrdering(true);`, I wasn't able to replicate the error.

Have you ensured that the publisher being created by the Autowired PubSubTemplate in your example is enabling message ordering?

**Comment by ls-urs-keller** [comment]
I have this property set in the code: https://github.com/ls-urs-keller/pubsub-message-ordering-problem/blob/main/src/test/java/com/lightspeedhq/pubsubmessageorderingproblem/PubSubOrderingIntegrationTest.java#L81
Maybe the bug is simply that this property is no longer picked up or was replaced by a different one.

**Comment by burkedavison** [comment]
I've created a ticket in the java-pubsub repository to see if someone from that team can take a look.

**Comment by burkedavison** [comment]
@ls-urs-keller : Feedback from the maintainers of Pub/Sub:

> This was an issue that required an update in the Pub/Sub emulator that was fixed in version 0.8.12 of the Pub/Sub emulator. Could you ask the customer to upgrade their version of the emulator?

**Comment by ls-urs-keller** [comment]
@burkedavison Thank you. I can confirm bumping the emulator container to `gcr.io/google.com/cloudsdktool/google-cloud-cli:467.0.0-emulators` or higher solves the problem.
